File: Answer/1984.学生分数的最小差值.go
import "sort"

/*
 * @lc app=leetcode.cn id=1984 lang=golang
 *
 * [1984] 学生分数的最小差值
 */

// @lc code=start
func minimumDifference(nums []int, k int) int {
	if k == 1 {
		return 0
	}
	sort.Ints(nums)
	ans := nums[k-1] - nums[0]
	for i := 1; i <= len(nums)-k; i++ {
		ans = min(ans, nums[i+k-1]-nums[i])
	}
	return ans
}

// @lc code=end

File: Answer/1984.学生分数的最小差值.py
#
# @lc app=leetcode.cn id=1984 lang=python3
#
# [1984] 学生分数的最小差值
#

# @lc code=start
from typing import List


class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        if k == 1:
            return 0
        nums.sort()
        ans = nums[k - 1] - nums[0]
        for i in range(1, len(nums) - k + 1):
            ans = min(ans, nums[i + k - 1] - nums[i])
        return ans
        
# @lc code=end

